src/main/kotlin/step2/calculator/OperationType.kt

```diff
@@ -1,50 +1,30 @@
 package step2.calculator
 
-enum class OperationType(private val operator: String) {
-    PLUS("+") {
-        override fun calculate(t: Double, u: Double): Double = t + u
-    },
-    MINUS("-") {
-        override fun calculate(t: Double, u: Double): Double = t - u
-    },
-    MULTIPLY("*") {
-        override fun calculate(t: Double, u: Double): Double = t * u
-    },
-    DIVIDE("/") {
-        override fun calculate(t: Double, u: Double): Double {
-            validateDivisor(u)
-            return t / u
-        }
-
-        private fun validateDivisor(u: Double) {
-            if (isZeroDivisor(u)) {
-                throw IllegalArgumentException(ZERO_DIVISOR_ERROR_MESSAGE)
-            }
-        }
-
-        private fun isZeroDivisor(u: Double) = u == 0.0
-    };
-
-    abstract fun calculate(t: Double, u: Double): Double
+enum class OperationType(private val operator: String, val calculate: (Double, Double) -> Double) {
+    PLUS("+", { leftOperand, rightOperand ->
+        leftOperand + rightOperand
+    }),
+    MINUS("-", { leftOperand, rightOperand ->
+        leftOperand - rightOperand
+    }),
+    MULTIPLY("*", { leftOperand, rightOperand ->
+        leftOperand * rightOperand
+    }),
+    DIVIDE("/", { leftOperand, rightOperand ->
+        validateDivisor(rightOperand)
+        leftOperand / rightOperand
+    });
 
     companion object {
-        private const val INVALID_OPERATOR_ERROR_MESSAGE = "옳바르지 않은 연산자입니다. 사칙연산에 해당하는 연산자만 입력하세요 :  %s"
-        private const val OPERATION_REGULAR = "^[+\\-*/\\d]*$"
+        private const val INVALID_OPERATOR_ERROR_MESSAGE = "올바르지 않은 연산자입니다. 사칙연산에 해당하는 연산자만 입력하세요 :  %s"
         private const val ZERO_DIVISOR_ERROR_MESSAGE = "0으로 나눌수 없습니다."
 
-        fun match(operator: String): OperationType {
-            return values().first { it ->
-                it.operator == operator.also { validateOperator(it) }
-            }
-        }
-
-        private fun validateOperator(operator: String) {
-            if (isNotContainOperator(operator)) {
-                throw IllegalArgumentException(String.format(INVALID_OPERATOR_ERROR_MESSAGE, operator))
-            }
-        }
+        fun match(operator: String): OperationType =
+            requireNotNull(
+                values().firstOrNull { it.operator == operator }
+            ) { INVALID_OPERATOR_ERROR_MESSAGE.format(operator) }
```

Comment:
elvis operator를 활용해볼 수도 있겠네요 :)
```suggestion
        fun match(operator: String): OperationType = values().find{ it.operator == operator }
            ?: throw IllegalArgumentException(INVALID_OPERATOR_ERROR_MESSAGE.format(operator))
```